reduction/fit_pointing_model.py

```diff
@@ -148,45 +156,62 @@ def quiver_segments(delta_az, delta_el, scale):
     theta2, r2 = np.arctan2(y2, x2), np.sqrt(x2 ** 2 + y2 ** 2)
     return np.c_[np.c_[theta1, r1], np.c_[theta2, r2]].reshape(-1, 2, 2)
 
+
 def param_to_str(model, p):
     """Represent value of *p*'th parameter of *model* as a string."""
     parameter = [param for param in model][p]
+    # Zeros are not displayed, which declutters the user interface
+    if parameter.value == 0.0:
+        return ''
     # Represent P9 and P12 (scale parameters) in shorter form
-    return parameter.value_str if p not in [8, 11] else ("%.3e" % parameter.value)
+    elif p in [8, 11]:
+        return "%.3e" % (parameter.value,)
+    else:
+        return parameter.value_str
+
 
 def update(fig):
     """Fit new pointing model and update plots."""
     # Perform early redraw to improve interactivity of clicks (which typically change state of target dots)
     # Target state: 0 = flagged, 1 = unflagged, 2 = highlighted
     target_state = keep * ((target_index == fig.highlighted_target) + 1)
     # Specify colours of flagged, unflagged and highlighted dots, respectively, as RGBA tuples
-    dot_colors = np.choose(target_state, np.atleast_3d(np.vstack([(1,1,1,1), (0,0,1,1), (1,0,0,1)]))).T
+    dot_colors = np.choose(target_state, np.atleast_3d(np.vstack([(1, 1, 1, 1), (0, 0, 1, 1), (1, 0, 0, 1)]))).T
     for ax in fig.axes[:7]:
         ax.dots.set_facecolors(dot_colors)
     fig.canvas.draw()
 
     # Fit new pointing model and update results
+    old_params = np.array(old_model.values())
+    # Zero the parameters marked as ZEROED but not the ones that are FIXED (or ENABLED)
+    old_params[parameter_state == ZEROED] = 0.0
+    new_model.set(old_params)
     params, sigma_params = new_model.fit(az[keep], el[keep], measured_delta_az[keep], measured_delta_el[keep],
-                                         std_delta_az[keep], std_delta_el[keep], enabled_params)
+                                         std_delta_az[keep], std_delta_el[keep], parameter_state == ENABLED,
+                                         keep_disabled_params=True)
```

Comment:
I get an error "TypeError: fit() got an unexpected keyword argument 'keep_disabled_params'"

Reply:
This goes away when I use the appropriate katpoint branch "spr1-407-fixed-parameters-in-pointing-model"

Reply:
You need to test this in tandem with ska-sa/katpoint#73, which has not been merged yet.